internal/server/proto/server.proto

```diff
@@ -2393,6 +2413,60 @@ message Artifact {
   google.protobuf.Any artifact = 1;
 }
 
+/********************************************************************
+* Ondemand Runners
+********************************************************************/
+message OndemandRunner {
+  // id is the unique ID of the runner config
+  string id = 1;
+
+  // oci_url is the OCI image that will be used to boot the ondemand runner.
+  string oci_url = 2;
+
+  // environment_variables is any env vars that should be exposed to the
+  // ondemand runner. This does not need to include any WAYPOINT specific
+  // variables, those are automatically added.
+  map<string, string> environment_variables = 3;
+
+  // plugin type is used to launch the plugin to start the batch task.
+  string plugin_type = 4;
+
+  // plugin config is the configuration for the plugin that is created. It
+  // is usually HCL and is decoded like the other plugins, and is plugin
+  // specific.
+  bytes plugin_config = 5;
+
+  // config format specifies the format of plugin_config
+  Project.Format config_format = 6;
```

Comment:
Speaking out loud again, not critical to this PR: we're using this in enough places now that maybe we should make this a standalone `HCLFormat` message/enum.